pkgs/by-name/sr/srm-cuarzo/package.nix

```
@@ -0,0 +1,63 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  meson,
+  ninja,
+  pkg-config,
+  libdisplay-info,
+  libdrm,
+  libGL,
+  libinput,
+  mesa,
+  seatd,
+  udev,
+}:
+stdenv.mkDerivation (self: {
+  pname = "srm-cuarzo";
+  version = "0.4.0-1";
+  rev = "v${self.version}";
+  hash = "sha256-PWtDSBYw62sfyN2lpd7NY8SbqDrt5IN1rdLZRtDfals=";
+
+  src = fetchFromGitHub {
+    inherit (self) rev hash;
```

```suggestion

  src = fetchFromGitHub {
    rev = "v${self.version}";
    hash = "sha256-PWtDSBYw62sfyN2lpd7NY8SbqDrt5IN1rdLZRtDfals=";
```
Isn't this a bit of unneeded recursion since they are only used once anyways?

---

It allows you to easily change them with overrideAttrs: `louvre.overrideAttrs (self: { rev = "..."; hash = "..."; })`

---

Is this a use case you often have? We usually don't do this and if you want to overwrite it anyways you should also be able to do:
```nix
  louvre.overrideAttrs (self: { src = pkgs.fetchFromGitHub {
    inherit (self.src) owner repo    
    rev = "..."; 
    hash = "...";
  }; })
```

---

It's something I do often enough (with other packages) for it to bug me that it takes this much noise to do. These two attrs is something that takes almost no effort to add and none to maintain but removes a lot of friction for what I would estimate to be one of the most common overrides for a package, which is changing the version (putting rev there is kind of for completeness's sake since that and hash belong together more than version and hash). And it doesn't take anything away because you can still override src altogether.

> We usually don't do this

Be the change you wish to see etc etc :^) There's a couple examples of at least having `rev` in a let..in block at the top of the file, though. Which is probably pre-recursive mkDerivation, but still.

FWIW I just tried something though, it looks like you can do this:

`overrideAttrs (self: { src = self.src.override { rev = "..."; hash = "..."; }; })`

I still don't really like it, but it's better than having to restate the whole fetchFromGitHub function with args.